src/Makefile

```diff
@@ -16,6 +16,8 @@ endif
 
 CFLAGS ?= -g -O2 -Werror -Wall
 ALL_CFLAGS += $(CFLAGS)
+ALL_CFLAGS += $(shell command -v pkg-config > /dev/null 2>&1 && pkg-config --cflags libelf)
```

Comment:
OK, let's do it so that it works for Debian and other distro folks, but still works for Facebook build system, that doesn't use pkg-config. Since number of distro folks will likely be bigger than number of others, we can keep pkg-config stuff by default, but there has to be an option to disable it so that it doesn't affect Facebook stuff, i.e. it can be declared explicitly "don't use pkg-config" rather then checking for pkg-config presense.

Let's introduce an option `NO_PKG_CONFIG` (disabled by default) and if option is provided that just hard-code `-lelf`.

Also, according to [1] pkg-config should not be used directly and should be overridable, i.e. there should be `$(PKG_CONFIG)` var that user may override.

It can be something like this:
```
ifndef NO_PKG_CONFIG
        ALL_CFLAGS += $(shell $(PKG_CONFIG) --cflags libelf)
endif
```
That way build system has full control over how pkg-config is used:
* if it wants to avoid it explicitly, just set `NO_PKG_CONFIG`;
* if it must be used, the use it and fail if pkg-config is not present.

that will provide deterministic build results.

[1] https://www.gnu.org/software/make/manual/html_node/Utilities-in-Makefiles.html#Utilities-in-Makefiles

Reply:
Sounds like a great idea! Pushed a new version implementing this and the other comment. The only difference is that given PKG_CONFIG is not defined by default, I've added: `PKG_CONFIG ?= pkg-config`